hhs_oauth_server/request_logging.py

```diff
@@ -60,23 +62,33 @@ def __str__(self):
         elif self.response.content:
             log_msg['size'] = len(self.response.content)
 
-        log_msg['user'] = str(get_user_from_request(self.request))
+        user = get_user_from_request(self.request)
+        log_msg['user'] = str(user)
         log_msg['ip_addr'] = get_ip_from_request(self.request)
 
         access_token = getattr(self.request, 'auth', get_access_token_from_request(self.request))
 
         if AccessToken.objects.filter(token=access_token).exists():
             at = AccessToken.objects.get(token=access_token)
-            log_msg['app_name'] = at.application.name
-            log_msg['app_id'] = at.application.id
-            log_msg['dev_id'] = at.application.user.id
-            log_msg['dev_name'] = str(at.application.user)
+            application = at.application
+            log_msg['app_name'] = application.name
+            log_msg['app_id'] = application.id
+            log_msg['dev_id'] = application.user.id
+            log_msg['dev_name'] = str(application.user)
             try:
-                log_msg['org_name'] = at.application.user.user_profile.organization_name
+                log_msg['org_name'] = application.user.user_profile.organization_name
             except Exception:
                 pass
             log_msg['access_token_hash'] = hashlib.sha256(str(access_token).encode('utf-8')).hexdigest()
 
+            # Update Application activity metric datetime fields
+            if application:
+                if user:
+                    application.last_active = timezone.now()
```

Comment:
this kind of behavior doesn't make sense in a logger, is there some other hook or middleware that might make more sense?

Reply:
TY for help finding a good location for this. Placed it in the OAuth2ResourceOwner() so that it is updated for any successful token authorization for the endpoints.